src/components/App.tsx

```diff
@@ -1762,6 +1779,39 @@ class App extends React.Component<AppProps, AppState> {
     this.setState({});
   };
 
+  /**
+   * @returns whether the menu was toggled on or off
+   */
+  public toggleMenu = (
+    type: "library" | "customSidebar",
+    force?: boolean,
+  ): boolean => {
+    if (type === "customSidebar" && !this.props.renderSidebar) {
+      console.warn(
+        `attempting to toggle "customSidebar", but no "props.renderSidebar" is defined`,
+      );
+      return false;
+    }
+
+    if (type === "library" || type === "customSidebar") {
+      const nextValue =
+        force !== undefined
```

Comment:
do we need specific check for `undefined` ?

Reply:
yep. If force is not undefined, we want to always set to the `force` flag. If it's undefined, we want to toggle.

Reply:
the naming may not be the most fortunate. I cargo culted it from classList.toggle.

Reply:
```suggestion
       let nextValue = null;
       if (force === undefined) {
         nextValue = this.state.openSidebar === type
          ? null
          : type;
        } else if (force) {
          nextValue = type;
        }
```
I think nested ternary operators do impact readability, I would suggest doing something similar as the above. for better readability / simplicity

Reply:
ternaries can be readable fine, but prettier's ternary styling is terrible and unreadable indeed.